censys/asm/inventory.py

```diff
@@ -57,7 +62,24 @@ def search(
         if fields:
             args["fields"] = fields
 
-        return self._get(self.base_path, args=args)
+        page = 1
+        next_cursor = None
+        hits = []
+        resp = self._get(self.base_path, args=args)
+        next_cursor = resp.get("nextCursor")
+        hits.extend(resp.get("hits", []))
+        while next_cursor and (page == 0 or pages == -1 or page < pages):
```

Comment:
Also - I think that this might result in getting the last "page" over and over again if they request a number of pages that does not return the full set. 

In those instances, a `nextToken` will still appear in the response, even though we hit our page limit.

Reply:
the condition is `next_cursor` AND `page < pages`, so even if there's a next cursor, if we have reached the page limit, it will stop

Reply:
you can run something like `censys asm search --query '*' --pages 3 --page-size 1` to test this